sdk/src/main/scala/io/horizen/account/state/AccountState.scala

```diff
@@ -465,8 +466,9 @@ class AccountState(
 
     ethTx.semanticValidity()
 
-    if (FeeUtils.GAS_LIMIT.compareTo(ethTx.getGasLimit) < 0)
-      throw new IllegalArgumentException(s"Transaction gas limit exceeds block gas limit: tx gas limit ${ethTx.getGasLimit}, block gas limit ${FeeUtils.GAS_LIMIT}")
+    val feeFork = GasFeeFork.get(stateMetadataStorage.getConsensusEpochNumber.getOrElse(0))
```

Comment:
just a question here, this `validate` is not tested during the block forging, so the next scenario:
- the fork is configured to happen at epoch 5 
- create a transaction, that is valid before the fork, but invalid after the fork
- this transaction is added to the mempool right before the first block of epoch 5 is about to be forged
- this transaction is added to the block which is in epoch 5

is it ok for this fork? I don't see any problems, as this tx also won't be validated when the block is send to other nodes, but wanted to double-check

Reply:
The TX would not be accepted in the block but will stay in the mempool, It eventually will be evicted according to the mempool policy after some time.